src/objectExplorer/objectExplorerService.ts

```diff
@@ -310,7 +317,7 @@ export class ObjectExplorerService {
      * OE out of
      * @param connectionCredentials Connection Credentials for a node
      */
-    public async createSession(promise: Deferred<vscode.TreeItem>, connectionCredentials?: IConnectionCredentials): Promise<void> {
+    public async createSession(promise: Deferred<vscode.TreeItem>, connectionCredentials?: IConnectionCredentials): Promise<string> {
```

Comment:
the promise should be `Deferred<vscode.TreeItem | undefined>` and the return type `Promise<string | undefined>`